plugins/TwoFactorAuth/tests/UI/TwoFactorAuth_spec.js

```diff
@@ -280,10 +302,17 @@ describe("TwoFactorAuth", function () {
         expect(await page.screenshotSelector('.loginSection,#content,#notificationContainer')).to.matchImage('twofa_forced_step2');
     });
 
+    it('should force user to setup 2fa when not set up yet but enforced step 2 show codes', async function () {
+        await page.click('.setupTwoFactorAuthentication .showOtpCodes');
+        await page.waitForSelector('.modal.open', {visible: true});
+        await page.waitForTimeout(250);
+
+        expect(await page.screenshotSelector('.loginSection,#content,#notificationContainer')).to.matchImage('twofa_forced_step2_showcodes');
```

Comment:
Could be better to only take a screenshot of the modal, rather then the other elements. That ensures that the UI test only needs to be updated when the modal content changes.